### PR TITLE
Merge 'dev' branch to 'master'

### PR DESCRIPTION
**Pull request by mitko501**
Pull request with changes from release 1.1

Major updates
- Block cipher Blowfish added
- Testsuite added
  - Tests for common streams (true stream, sac streams ....)
  - Tests for sha3 streams (tests vectors for most common used functions)
  - Tests for estream streams (tests vectors for most common used functions)

### CODE REVIEW

File: testsuite/test-utils/estream_test_case.h [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+         * {plaintext2 in hex}
+         * {key2 in hex}
+         * {initialization vector2}
+         * {ciphertext2}
+         *
+         * STRUCTURE OF CIPHERTEXT
+         * two options
+         * - whole plaintext
+         * - tuples (index where ciphertext begins) (ciphretext)
+         *     for example "0 FF 5 EE" means that first byte is FF and sixth byte is EE
```
Comment (Bender250) [inline testsuite/test-utils/estream_test_case.h]: Is this useful?

File: testsuite/test-utils/estream_test_case.h [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+         *
+         * @param input stream
+         * @param test_case test_case instance
+         * @return input stream
+         */
```
Comment (Bender250) [inline testsuite/test-utils/estream_test_case.h]: Nice, can you add this description, how to write new tests and how to run tests to [wiki](https://github.com/crocs-muni/eacirc-streams/wiki)? Thanks

File: testsuite/test-utils/sha3_test_case.h [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+         * two options
+         * - whole plaintext
+         * - tuples (index where ciphertext begins) (ciphretext in hex)
+         *     for example "0 FF 5 EE" means that first byte is FF and sixth byte is EE
+         *
+         *
+         * @param input stream
+         * @param test_case test_case instance
+         * @return input stream
+         */
```
Comment (Bender250) [inline testsuite/test-utils/sha3_test_case.h]: Are the docs of `std::istream &operator>>` copy&paste? Shouldn't they be moved to parent definition?

File: testsuite/test-utils/estream_test_case.h (hunk 2 above)
Comment (mitko501) [reply testsuite/test-utils/estream_test_case.h]: Created issue https://github.com/crocs-muni/eacirc-streams/issues/14

File: testsuite/test-utils/sha3_test_case.h (hunk 3 above)
Comment (mitko501) [reply testsuite/test-utils/sha3_test_case.h]: No it is not copy&paste sha3 has different test vector than estream. Estream has key and plaintext and sha3 only plaintext

**Pull request merged**